### ISSUE TITLE
Can't save any track information

### ISSUE THREAD

**Issue by Jujucristine** [clementine-player/Clementine#4833]
This has been going on since I downloaded clementine months ago, on version 1.2.1 I think. I now have version 1.2.3 installed, on windows version.

One of the features I loved the most on Amarok was is rating system, and this, among other things, simply does not work at all on Clementine. It seems it can't save any track information at all, on the main tab (on the "Edit tags" tab you can change things just fine). It can't save play counts, rating, or anything else, even going on edit, then save, it never saves anything. I've already tried running it as administrator, but the problem persists.

So, basically, anything that displays when going to edit track information can't be edited or changed, even by the automatic track count system itself.

**Comment by rgould** [comment]
:+1: I'm experiencing this as well

**Comment by Chocobozzz** [comment]
Please try with the latest build http://builds.clementine-player.org/ and tell us if the bug is still here.

**Comment by rgould** [comment]
@Chocobozzz Just tried it with `clementine-1.2.3-1174-g971df03.dmg`, and it seems to still be there. 

I'll throw an album into a playlist, select all, click `Complete tags automatically...`, and once that's done, the playlist looks okay.

If I then play a song, after it's done playing (or I skip to the next song), the metadata in the files hasn't changed at all, and the playlist then reflects this. I'm working with mp3s here.

**Comment by Jujucristine** [comment]
Nothing changed for me either with clementine-1.2.3-1174-g971df03.exe.

**Comment by ahills** [comment]
I have this issue for MP3 and FLAC files on OS X with the latest build (http://builds.clementine-player.org/mac/clementine-1.2.3-1249-gd886609.dmg).

**Comment by Chocobozzz** [comment]
Please provide some logs by run Clementine from the command line

**Comment by ahills** [comment]
```
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
TagLib: FLAC::File::save() - Cannot save to a read only file.
```
After changing the permissions, it worked fine. It would be nice if the UI would give some feedback.

**Comment by Jujucristine** [comment]
Which permissions did you change? None of my music files are marked read only, and it doesn't work running clementine as administrator.

**Comment by ahills** [comment]
I just ran `chmod -R u+w ~/Music` and that took care of it. I don't know how OS X handles file permissions beyond the classic Unix modes.

**Comment by Jujucristine** [comment]
Oh, I'm on windows. The folder (but not the files inside) were actually set to read only, but unchecking that and applying to all sub folders didn't help, unfortunately. It still does not work.

(And windows command prompt is horrible, no idea how to make it run clementine there. Tried some things but even when it "runs" it just opens another window written clementine in it. Would be nice if clementine had a logs inside the program itself)

**Comment by TheUbuntuGuy** [comment]
We have debug builds which have a console. You can download it from here: http://builds.clementine-player.org/win32/debug/ (Just use the top one)

**Comment by ahills** [comment]
Find the path of the Clementine executable and type the whole thing into the command prompt surrounded by quotes, e.g., `"\Program Files\Clementine\clementine.exe"`. It should just launch the regular Clementine application as usual. Try to make your changes, and then watch the command line window, which should still be open.

**Comment by Jujucristine** [comment]
Yes, I forgot you can use cd on windows as well, since starting from the default path you can't do that. (I never bother with windows command prompt unless there's no other way to something...), but it can't display any messages, or none show up.

From the debug version, when trying to save some rating info:
14:37:21.256 DEBUG _Mess14:3ageReplyBase:24 7:21.260  DEBUG  MessageReply<Messag
eType >:90          Re   leasing  ID Waiting 10 (finished) on
ID 10
14:37:21.282 DEBUG _MessageReplyBase:26             Acquired ID 10
14:37:21.291 DEBUG _Me14:37:21.293 ssageReplyBase:24 DEBUG  MessageReply<Message
Type>:90         Releasing  ID  11  (finished)
   Waiting on ID 11
14:37:21.304 DEBUG _MessageReplyBase:26             Acquired ID 11
14:37:21.309 DEBUG PlaylistBackend:294              Saving playlist 1
14:37:25.360 DEBUG _MessageReplyBase:24 14:37:25.364    DEBUG  MessageRepl y<Mes
sageT y pe > : 90     W aiting  Releasing onID 12  (finished) ID
12
14:37:25.389 DEBUG _MessageReplyBase:26             Acquired ID 12
14:37:46.735 DEBUG _MessageReplyBase:24    14:37:4 6.7 44  DEBUG  MessageReply<M
ess a geType>:90       Waiting  on Releasing ID ID 13 13
(finished)

When finishing playing a track (which should modify play count):
"file:///C:/Users/user/Desktop/Jujucristine/M·sicas/Shadow of the colossus
/17 Wander's Death.mp3" )  to "Koh Ohtani" "Wander's Death"
14:40:43.732 DEBUG _MessageReplyBase:24             Waiting on ID 15
14:40:43.779 INFO  MoodbarLoader:108                Loading cached moodbar data
for "C:/Users/user/Desktop/Jujucristine/M·sicas/Rurouni Kenshin OST/Ruroun
i Kenshin OST III Kyoto Kessen/06-Warriors Blue -A Theme of Shinomori Aoshi-.mp3
"
14:40:43.803 DEBUG PlaylistBackend:294              Saving playlist 1
14:40:43.816 DEBUG MessageReply<MessageType>:90     Releasing ID 15 (finished)
14:40:43.823 DEBUG _MessageReplyBase:26             Acquired ID 15
14:40:43.874 DEBUG MainWindow:1361                  position 0 scrobble point 60
 status 0

**Comment by TheUbuntuGuy** [comment]
Clementine isn't logging anything out of the ordinary there.

I tested this on my Windows machine. The tags do indeed get written to the files correctly (I've tested MP3 and FLAC), but the SQLite database is not updating with the new tags/ratings.

**Comment by Jujucristine** [comment]
Another thing I noticed is that when I went to check the files, I changed the rating on one of them directly in the file. When trying to edit it on clementine later, the rating displays as saved, but the rating row stays empty.

**Comment by DegreesK0** [comment]
It's been 7 years so i don't know if you still have this problem, but might help others who get stuck.

Installed clementine on a fresh install of Ubuntu. Started playing music and, like Jujucrristine, tried to change cover art and ratings but nothing stuck. After scouring the internet and trying a ton of things i gave up and started loading all my music files into clementine. And then i noticed i could make edit changes and it workred

To clarify, loading music files was done by opening up preferences(Ctrl+P), selecting Music library and clicking add new folder before choosing the folder with said music.

I think the issue might have to do with how clementine observes permissions for editing your music files

**Comment by docvalentine** [comment]
I'm having this problem on Steam Deck right now. I can make changes and they appear visually, but as soon as I play those tracks, they revert.

All of my music is loaded as DegreesK0 suggested, by adding library folders and scanning.
